span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToLogRecordsTransformer.java

```diff
@@ -0,0 +1,72 @@
+package org.hypertrace.core.spannormalizer.jaeger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.util.Timestamps;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.Span;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.hypertrace.core.datamodel.Attributes;
+import org.hypertrace.core.datamodel.LogEvent;
+import org.hypertrace.core.datamodel.LogEvents;
+import org.hypertrace.core.spannormalizer.util.AttributeValueCreator;
+
+public class JaegerSpanToLogRecordsTransformer
+    implements Transformer<byte[], PreProcessedSpan, KeyValue<String, LogEvents>> {
+
+  @Override
+  public void init(ProcessorContext context) {
+    // no-op
+  }
+
+  @Override
+  public KeyValue<String, LogEvents> transform(byte[] key, PreProcessedSpan preProcessedSpan) {
+    Span value = preProcessedSpan.getSpan();
+    String tenantId = preProcessedSpan.getTenantId();
+    if (value.getLogsCount() == 0) {
+      return null;
+    }
+
+    return new KeyValue<>(null, buildLogEventRecords(value, tenantId));
+  }
+
+  @VisibleForTesting
+  LogEvents buildLogEventRecords(Span value, String tenantId) {
```

Comment:
Pl. wrap this in try/catch for any runtime errors so, and log the debug message, so it doesn't break the entire span-normalizer.